# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Gunner2033d/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why
#Lane 2: Refresh / Content Opportunity Scoring

*I am choosing this lane because I want to investigate how observable page-level signals can be used to prioritize content for review. The starter data includes measures such as impressions, CTR, average position, content age, and trend direction, giving me several signals to explore when deciding which pages may deserve attention. This is a provisional direction, and I will test whether these signals can produce a useful ranking for human review.*

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

#Decision
*The decision I want to improve is which pages should receive review priority when content teams have limited time.*
#Action
*The action is for a reviewer to inspect the highest-priority pages and decide whether to refresh, improve, protect, or monitor them.*
#cost of wrong call
*A wrong recommendation could waste time on a page that does not need attention, while a missed opportunity could leave a useful page without review. Data and ML may help by combining multiple observable signals into a consistent prioritization method.*

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import os
import sys
import subprocess
import pandas as pd

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))

corr = df["avg_position"].corr(df["ctr"])
print(f"avg_position vs ctr correlation: {corr:.3f}")

age_by_trend = df.groupby("trend_direction")["content_age_days"].median()
print("\nMedian content age by trend direction:")
print(age_by_trend.round(0).to_string())


Total pages: 30000
avg_position vs ctr correlation: -0.073

Median content age by trend direction:
trend_direction
down      216.0
flat      231.0
new       279.0
stable    300.0
up        292.0


#Dataset Observation


The starter dataset contains **30,000 pages**. The correlation between `avg_position` and `ctr` is **-0.073**, showing a weak negative relationship. Median content age is **216 days for declining pages** compared with **292 days for pages trending up**. These observations provide enough variation to investigate whether multiple signals can help prioritize pages for review.


## 4. Careful words: what I can and can't claim

*I can use the available signals to identify patterns and build a ranking that supports human review. I can describe the results as observed or directional relationships. I cannot claim that a particular signal causes a page to decline, that a refresh will definitely improve performance, or that the model explains Google's ranking algorithm.*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.